File: prak2 java/Memory.java
class Memory extends Hardware{
    
    private String frequency;
    private String memorySize;
    private String supportsCuda;

    Memory(){
        //konstruktor
    }

    Memory(Hardware x, String frequency, String memorySize, String supportsCuda){
        //konstruktor
        this.setPrice(x.getPrice());
        this.setIdProduct(x.getIdProduct());
        this.setBrand(x.getBrand());
        this.setModel(x.getModel());
        this.frequency = frequency;
        this.memorySize = memorySize;
        this.supportsCuda = supportsCuda;
    }

    Memory(int price, String idProduct, String brand, String model, String frequency, String memorySize, String supportsCuda){
        //konstruktor
        this.setPrice(price);
        this.setIdProduct(idProduct);
        this.setBrand(brand);
        this.setModel(model);
        this.frequency = frequency;
        this.memorySize = memorySize;
        this.supportsCuda = supportsCuda;
    }

    //get set
    public String getFrequency() {
        return frequency;
    }

    public void setFrequency(String frequency) {
        this.frequency = frequency;
    }

    public String getMemorySize() {
        return memorySize;
    }

    public void setMemorySize(String memorySize) {
        this.memorySize = memorySize;
    }

    public String getSupportsCuda() {
        return supportsCuda;
    }

    public void setSupportsCuda(String supportsCuda) {
        this.supportsCuda = supportsCuda;
    }

    //methods
    public void tampilkanMemory(){
        System.out.println("Price: " + getPrice());
        System.out.println("idProduct: " + getIdProduct());
        System.out.println("Brand: " + getBrand());
        System.out.println("Model: " + getModel());
        System.out.println("Frequency: " + getFrequency());
        System.out.println("MemorySize: " + getMemorySize());
        System.out.println("SupportsCuda: " + getSupportsCuda());
    }

}

File: prak2 c++/Memory.cpp
#include <iostream>
#include <string>

using std::string;
using std::cout;
using std::endl;

class Memory : public Hardware{

    private:
        string frequency;
        string memorySize;
        string supportsCuda;

    public:
        Memory(){
            //konstruktor
        }

        Memory(Hardware x, string fr, string me, string su){
            //konstruktor
            this->setPrice(x.getPrice());
            this->setIdProduct(x.getIdProduct());
            this->setBrand(x.getBrand());
            this->setModel(x.getModel());
            frequency = fr;
            memorySize = me;
            supportsCuda = su;
        }

        Memory(int pr, string id, string br, string mo, string fr, string me, string su){
            //konstruktor
            this->setPrice(pr);
            this->setIdProduct(id);
            this->setBrand(br);
            this->setModel(mo);
            frequency = fr;
            memorySize = me;
            supportsCuda = su;
        }


        //set
        void setFrequency(string fr){
            frequency = fr;
        }

        void setMemorySize(string me){
            memorySize = me;
        }

        void setSupportsCuda(string su){
            supportsCuda = su;
        }

        //get
        string getFrequency(){
            return frequency;
        }

        string getMemorySize(){
            return memorySize;
        }

        string getSupportsCuda(){
            return supportsCuda;
        }


        //methods
        void tampilkanMemory(){
            cout << "Price : " << getPrice() << endl;
            cout << "idProduct : " << getIdProduct() << endl;
            cout << "Brand : " << getBrand() << endl;
            cout << "Model : " << getModel() << endl;
            cout << "Frequency : " << getFrequency() << endl;
            cout << "MemorySize : " << getMemorySize() << endl;
            cout << "SupportsCuda : " << getSupportsCuda() << endl;
        }


        ~Memory(){
            //destruktor
        }

};
